spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/LettuceConnectionConfiguration.java

```diff
@@ -146,6 +146,8 @@ public LettuceClientConfigurationBuilder createBuilder(Pool properties) {
 			config.setMaxTotal(properties.getMaxActive());
 			config.setMaxIdle(properties.getMaxIdle());
 			config.setMinIdle(properties.getMinIdle());
+			config.setTimeBetweenEvictionRunsMillis(
```

Comment:
Jedis supports the same configuration property. Would you like to make the equivalent change there as well?

Reply:
I think it does not the responsibility of Spring-Boot.
Spring-Boot configured the Jedis-Pool using the below code:

https://github.com/spring-projects/spring-boot/blob/36c1c051b8106371d37f8a747cea0e369a8e5c84/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/JedisConnectionConfiguration.java#L112

So, i think `JedisPoolConfig` is responsibility to `Jedis`.
`JedisPoolConfig` sets the default value of `TimeBetweenEvictionRunsMillis`

https://github.com/xetorthio/jedis/blob/ef4ab403f9d8fd88bd05092fea96de2e9db0bede/src/main/java/redis/clients/jedis/JedisPoolConfig.java#L10

So I'm not sure I should handle this equally.

I will wait for your opinion, Thank you for your fast feedback and thoughtful review :)

Reply:
I think it does not the responsibility of Spring-Boot.
Spring-Boot configured the Jedis-Pool using the below code:

https://github.com/spring-projects/spring-boot/blob/36c1c051b8106371d37f8a747cea0e369a8e5c84/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/JedisConnectionConfiguration.java#L112

So, i think `JedisPoolConfig` is responsibility to `Jedis`.
`JedisPoolConfig` sets the default value of `TimeBetweenEvictionRunsMillis`

https://github.com/xetorthio/jedis/blob/ef4ab403f9d8fd88bd05092fea96de2e9db0bede/src/main/java/redis/clients/jedis/JedisPoolConfig.java#L10

So I'm not sure I should handle this equally.

I will wait for your opinion, Thank you for your fast feedback and thoughtful review :)

Reply:
Thanks. I'd missed that `JedisPoolConfig` sets the property already and, therefore, has a different default to Lettuce.

Unfortunately, this creates a problem as the `Pool` class is shared between Jedis and Lettuce configuration properties. As currently proposed here, the changes will result in a `spring.redis.jedis.pool` property that has no effect. We need to figure out how to avoid that while also coping with the different defaults.

I think our best option is to declare the `timeBetweenEvictionRuns` property with a `null` default value and only map it onto the Lettuce or Jedis pool config if the value is non-null.

Reply:
I agree that it seems to be the best way now, I modified it.
Thank you for your clear and good feedback. 😄